integration/messaging/test/test-updateToken.js

```diff
@@ -0,0 +1,102 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const seleniumAssistant = require('selenium-assistant');
+const fetch = require('node-fetch');
+const expect = require('chai').expect;
+
+const setupNotificationPermission = require('./utils/setupNotificationPermission');
+const testServer = require('./utils/test-server');
+const getErrors = require('./utils/getErrors');
+const retrieveFCMToken = require('./utils/retrieveFCMToken');
+const timeForward = require('./utils/timeForward');
+const getFCMToken = require('./utils/getFCMToken');
+const deleteFCMToken = require('./utils/deleteFCMToken');
+const makeFCMAPICall = require('./utils/makeFCMAPICall');
+const getReceivedMessages = require('./utils/getReceivedMessages');
+const demoSetup = require('./utils/getDemoSetup');
+
+const ENDPOINT = 'https://fcm.googleapis.com';
+const DEMOS = demoSetup.DEMOS;
+const EIGHT_DAYS_IN_MS = 8 * 86400000;
+
+describe('Firebase Messaging Integration Tests > update a token', function() {
+  this.timeout(60 * 1000);
+  if (process.env.TRAVIS) {
+    this.retries(3);
+  } else {
+    this.retries(1);
+  }
+
+  let globalWebDriver;
+
+  async function cleanUp() {
+    await seleniumAssistant.killWebDriver(globalWebDriver);
+  }
+
+  before(async function() {
+    await testServer.start();
+  });
+
+  after(async function() {
+    await testServer.stop();
+    await cleanUp();
+  });
+
+  const availableBrowsers = seleniumAssistant.getLocalBrowsers();
+  availableBrowsers.forEach(assistantBrowser => {
+    // Only test on Chrome and Firefox
+    if (
+      assistantBrowser.getId() !== 'chrome' &&
+      assistantBrowser.getId() !== 'firefox'
+    ) {
+      return;
+    }
+
+    DEMOS.forEach(demoInfo => {
+      describe(`${assistantBrowser.getPrettyName()} : ${
+        demoInfo.name
+      }`, function() {
+        beforeEach(async function() {
+          await cleanUp();
+
+          assistantBrowser = setupNotificationPermission(
+            assistantBrowser,
+            testServer.serverAddress
+          );
+          globalWebDriver = await assistantBrowser.getSeleniumDriver();
+        });
+
+        it(`should update a token`, async function() {
+          await globalWebDriver.get(
+            `${testServer.serverAddress}/${demoInfo.name}/`
+          );
+
+          const token = await retrieveFCMToken(globalWebDriver);
+          expect(token).to.exist;
+
+          // roll the clock forward > 7days
+          await timeForward(globalWebDriver, EIGHT_DAYS_IN_MS);
+          try {
+            await getFCMToken(globalWebDriver);
```

Comment:
Related to the above question - this function is getting a toke, but it's not checking for anything so it's unlikely to be testing the behavior you think it is (i.e. `executeScript()` returns a promise, which I don't think is returning the promise you return from within the promise).

Reply:
I actually want to test the right network call is made, but not sure if there's a way to do that. Changed it to check that the token exists.